addons/mail/static/src/models/message/message.js

```diff
@@ -523,6 +648,156 @@ function factory(dependencies) {
             return replace(threads);
         }
 
+        /**
+         * Apllies the hightlighting based the thread search text. 
+         * 
+         * @private
+         * @param {String} content
+         * @returns {String}
+         */
+        _highlightBasedOnThreadSearch(content) {
+            if (this.threads) {
+                for (const thread of this.threads) {
+                    if (thread.searchedText) {
+                        content = content.replace(
+                            new RegExp(`(${thread.searchedText})`,"ig"),
+                            '<b class="o_MessageInlineHighlight">$1</b>'
```

Comment:
You can absolutely not do text-search over the html as text and expect it to work. Tag names, attributes and a whole lot of other stuff *will* match and break.

Reply:
@sdegueldre I agree. It was decided for a v1 that we didn't care, but that's just bad. Then the best solution I see is:
1) construction a DOMElement
2) Get all the text node and text content
3) Apply the transformation on those
4) Return the innerHTML from the constructed Element

Slower but only used when searching so I guess it's kinda alright 
Thoughts ?

Reply:
Sorry for the late reply, I think I missed the notification.

Yes, I think that's better, using a naive approach will probably have some limitations (eg no matches if the text is partially bold/italics) but at least it won't just break the existing markup. There are ways to improve this further later (eg by creating a string containing the full text content, and then finding match offsets in the string and mapping that back to DOM elements) but I think as a first approach it should be enough.

Still not a fan of going through html-as-text every time but I guess currently there's not really a good way around it.

Reply:
I went with a smarter regex cause the HTML manipulation solution came out to be more tedious than I though. I tried a few methods and each time the result was not good and the code quite complex.